src/services/sync/strategies/consumerActions/SchoolAction.js

```diff
@@ -27,6 +27,12 @@ class SchoolAction extends BaseConsumerAction {
 			return;
 		}
 
+		// don't create school as new if exists, but not set for ldap yet (not ready to migrate)
+		const checkedSchool = await SchoolRepo.findSchoolByOfficialSchoolNumber(schoolData.ldapSchoolIdentifier);
+		if (checkedSchool) {
+			return;
+		}
+
```

Comment:
Doesn't work the LDAP Strategies give not back an Official School Number.
Maybe for some LDAP Providers there is a correlation between the Official school number and the ldap id, but in most cases there is no correlation.
See there for
https://github.com/hpi-schul-cloud/schulcloud-server/blob/bc-1240-ldap-duplicate-school/src/services/ldap/strategies/general.js#L19
https://github.com/hpi-schul-cloud/schulcloud-server/blob/bc-1240-ldap-duplicate-school/src/services/ldap/strategies/iserv-idm.js#L21

Reply:
when there is no official school number, works as before: school is created

the problem is not to create schools, which have no ldapId yet (were not migrated) as new schools, when we have them in our db.